R/extend_scenario_trajectory.R

```diff
@@ -372,3 +384,92 @@ handle_phase_out_and_negative_targets <- function(data) {
       )
     )
 }
+
+#' Calculate the ratio of the required change in technology that each company
+#' has achieved per technology at the end of the production forecast period.
+#' This ratio will later serve to adjust the net profit margin for companies
+#' that have not built out enough production capacity in increasing technologies
+#' and hence need to scale up production to compensate for their lag in buildout.
+#'
+#' @param data A data frame containing the production forecasts of companies
+#'   (in the portfolio). Pre-processed to fit analysis parameters and after
+#'   conversion of power capacity to generation.
+#' @param start_analysis Numeric. A vector of length 1 indicating the start
+#'   year of the analysis.
+#' @param time_frame Numeric. A vector of length 1 indicating the number of
+#'   years for which forward looking production data is considered.
+#' @param baseline_scenario Character. A vector of length 1 indicating baseline
+#'   scenario
+#'
+#' @noRd
+calculate_proximity_to_target <- function(data,
+                                          start_analysis,
+                                          time_frame,
+                                          baseline_scenario) {
+  data <- data %>%
```

Comment:
this function throws warnings (no non-missing arguments to max; returning -Inf) at least for me locally, also I would argue it is unnecessarily complicated (unless I am overlooking a complexity).
How about:
- filtering data for the needed lines (year = .env$start_analysis + .env$time_frame, scenario = shock), 
- calculating proximity_to_target with some nice and tidy mutate statements
- right joining this to data